src/native/libs/System.Native/pal_networking.c

```diff
@@ -3143,6 +3143,11 @@ int32_t SystemNative_Disconnect(intptr_t socket)
 #elif HAVE_DISCONNECTX
     // disconnectx causes a FIN close on OSX. It's the best we can do.
     err = disconnectx(fd, SAE_ASSOCID_ANY, SAE_CONNID_ANY);
+    if (err != 0 && errno == EOPNOTSUPP)
```

Comment:
We could use the same condition here like we do with Linux: `if (err != 0)`.

Reply:
updated